### PR TITLE
Binary opt2

### PR DESCRIPTION
**Pull request by luojing1211**
This is new update for binary model based on Rutger's comment 1. There will be more updates, so don't merge yet.

### CODE REVIEW

**Comment by vhaasteren** [comment]
@luojing1211 can you fix the conflict? Just make sure that test_derivates ends with:

    parfile = os.path.join(datapath, 'B1855+09_NANOGrav_dfg+12_modified.par')
    timfile = os.path.join(datapath, 'B1855+09_NANOGrav_dfg+12.tim')
    ddm = mb.get_model(parfile)
    t = toa.get_TOAs(timfile,planets = True)
    ddob = ddm.get_dd_object(t.table)

File: pint/models/pint_pulsar_binary.py [hunk 1]
```diff
@@ -120,3 +144,10 @@ def binary_delay(self,toas):
         for bdf in self.binary_delay_funcs:
             bdelay+= bdf(toas)
         return bdelay
+
+    @Cache.use_cache
+    def d_delay_d_xxxx(self,param,toas):
+        """Return the DD timing model delay derivtives"""
```
Comment (vhaasteren) [inline pint/models/pint_pulsar_binary.py]: The comment to return the 'DD' binary model seems slightly wrong. DD is only the 'default' now, right?

File: pint/models/pint_dd_model.py [hunk 2]
```diff
@@ -19,48 +20,33 @@ class DDwrapper(PSRbinaryWapper):
...
-             units="second",
-             description="Binary Einsten delay GAMMA term",
-             parse_value=np.double),binary_param = True)
-
-        self.add_param(Parameter(name="DR",
-             units="",
-             description="Relativistic deformation of the orbit",
-             parse_value=np.double),binary_param = True)
+        self.binary_model_name = 'DD'
+        self.binary_model_class = DDmodel
```
Comment (vhaasteren) [inline pint/models/pint_dd_model.py]: So we are using the 'DD' model by default?

I know I have not updated the BT model, so definitely leave it for now, but is this going to be standard?

Comment (luojing1211) [inline pint/models/pint_dd_model.py]: This is only happened in pint_dd_model.py. So it is DD model specific. I believe DD model should not be the standard model.

File: pint/models/pint_pulsar_binary.py (hunk 1 above)
Comment (luojing1211) [inline pint/models/pint_pulsar_binary.py]: I have not modified derivative part now. Yes, I am going to change it .

**Pull request closed without merging**